### PR TITLE
chore: split ci into separate steps

### PR DESCRIPTION
**Pull request by as-herzog**
### Description

Splits up the `Build` step to have separate steps for `unit test`,`integration test`, and `build` for easier debugging. Also skips duplicate runs of each step due to our on push and on pull_request settings. 

**Before:**
Screen Shot 2021-05-14 at 3 42 37 PM

**After:**
Screen Shot 2021-05-14 at 4 00 10 PM

### Reference
https://github.community/t/duplicate-checks-on-push-and-pull-request-simultaneous-event/18012

### CODE REVIEW

**Comment by as-herzog** [comment]
I see that we only run Snyk on pull requests right now, do we need to run the others on push or could we just run all of them on PR?

File: .github/workflows/go.yml [hunk 1]
```diff
@@ -1,12 +1,13 @@
...
 jobs:
-
-  build:
-    name: Build
+  unit:
+    name: Unit Test
     runs-on: ubuntu-latest
+    # We want to run on external PRs, but not on our own internal PRs as they'll be run
+    # by the push to the branch.
+    if: github.event_name == 'push' || github.event.pull_request.head.repo.full_name != github.repository
```
Comment (cyx) [inline .github/workflows/go.yml:9]: 💪

File: .github/workflows/goreleaser.yml [hunk 2]
```diff
@@ -7,6 +7,9 @@ on:
 jobs:
   goreleaser:
     runs-on: ubuntu-latest
+    # We want to run on external PRs, but not on our own internal PRs as they'll be run
+    # by the push to the branch.
+    if: github.event_name == 'push' || github.event.pull_request.head.repo.full_name != github.repository
```
Comment (cyx) [inline .github/workflows/goreleaser.yml:12]: Shouldn't we also avoid running this for external PRs? I guess what's our full intent?

**Review by cyx — APPROVED** [review]
Just have a question re: releasing for external PRs

**Pull request merged**